mindmeld/active_learning/results_manager.py

```diff
@@ -0,0 +1,190 @@
+                based on the current timestamp.
+        """
+        strategies = "_".join(
+            STRATEGY_ABRIDGED[s] for s in selection_strategies if s in STRATEGY_ABRIDGED
+        )
+        now = datetime.datetime.now()
+        self.experiment_folder_name = (
+            f"{now.month}-{now.day}_{now.hour}:{now.minute}_{strategies}"
+        )
+
+    @property
+    def experiment_folder(self):
+        """
+        Returns:
+            experiment_folder (str): Path to the Active Learning experiment folder.
+        """
+        return os.path.join(self.output_folder, self.experiment_folder_name)
+
+    def create_experiment_folder(
+        self, active_learning_params: Dict, training_strategies: List
+    ):
+        """Creates the active learning experiment folder.
+        Args:
+            active_learning_params (Dict): Dictionary representation of the params to store.
+            training_strategies (list): List of strategies used for the experiment.
+        """
+        self.set_experiment_folder_name(training_strategies)
+        create_dir_if_absent(self.experiment_folder)
+        self.dump_json(AL_PARAMS_PATH, active_learning_params)
+        self.create_folder(AL_RESULTS_FOLDER)
+        self.create_folder(AL_PLOTS_FOLDER)
+
+    def create_folder(self, unformatted_path):
+        """Creates a folder given an unformatted path.
+        Args:
+            unformatted_path (str): Unformatted path to JSON file.
+        """
+        os.makedirs(self.format_path(unformatted_path))
+
+    def format_path(self, unformatted_path):
+        """
+        Args:
+            unformatted_path (str): Unformatted path to JSON file.
+        Returns:
+            formatted_path (str): Path formatted with the experiment folder.
+        """
+        return unformatted_path.format(experiment_folder=self.experiment_folder)
+
+    def load_json(self, unformatted_path: str):
+        """Load JSON data from file. If the JSON file doesn't exist, an empty json file is created.
+        Args:
+            unformatted_path (str): Unformatted path to JSON file.
+        Returns:
+            json_data (Dict): Loaded JSON data.
+        """
+        formatted_path = unformatted_path.format(
+            experiment_folder=self.experiment_folder
+        )
+        if not os.path.isfile(formatted_path):
+            self.dump_json(formatted_path, data={})
+        with open(formatted_path, "r") as infile:
+            json_data = json.load(infile)
+            infile.close()
+        return json_data
+
+    def dump_json(self, unformatted_path: str, data: Dict):
+        """Dump data to a JSON file.
+        Args:
+            unformatted_path (str): Unformatted path to JSON file.
+            data (Dict): Data to dump.
+        """
+        formatted_path = self.format_path(unformatted_path)
+        with open(formatted_path, "w") as outfile:
+            json.dump(data, outfile, indent=4)
+            outfile.close()
```

Comment:
I thought with the `with` clause, you dont have to `close()`?

Reply:
You're right, using .close() is redundant here. Removed it here and in data_loading.py